lib/daru/io/io.rb

```diff
@@ -201,25 +199,29 @@ def from_csv_prepare_opts opts
 
       def from_csv_hash_with_headers(path, opts)
         opts[:header_converters] ||= :symbol
-
         ::CSV
-          .read(path, 'rb',opts)
+          .parse(new_open(path), opts)
           .tap { |c| yield c if block_given? }
           .by_col.map { |col_name, values| [col_name, values] }.to_h
       end
 
       def from_csv_hash(path, opts)
         csv_as_arrays =
           ::CSV
-          .open(path, 'rb', opts)
+          .parse(new_open(path), opts)
           .tap { |c| yield c if block_given? }
           .to_a
-
         headers       = ArrayHelper.recode_repeated(csv_as_arrays.shift)
         csv_as_arrays = csv_as_arrays.transpose
-
         headers.each_with_index.map { |h, i| [h, csv_as_arrays[i]] }.to_h
       end
+
+      def new_open(url)
+        open(url)
+      rescue URI::InvalidURIError
+        host = url.match(".+\:\/\/([^\/]+)")[1]
+        Net::HTTP.get host, url.partition(host)[2] || '/'
```

Comment:
@athityakumar I believe, this approach is redundand. 
Daru is pretty complicated piece of software, and each new complication should be taken with a grain of salt (also known as YAGNI). But we now have in the core of library additional method which targets:
a) rare cases (underscored domains);
b) ...for outdated Ruby version (Daru currently supports 2.0+, yet we'll drop 2.0 soon, and 2.1 in a few months, and in 2.2 the problem is fixed already);
c) ...without much explanations.

Considering all of this, I'd say just drop this trick completely. If we'll have an issue saying "this VERY IMPORTANT research data I need is on obscure_domain.some_large_univercity.com", we'll just answer "It is known Ruby bug, just download data manually or update your Ruby version", and that would be it.